### ISSUE TITLE
bmat error: reshape() takes from 1 to 2 positional arguments but 3 were given

### ISSUE THREAD

**Issue by skoudoro** [cvxpy/cvxpy#2548]
**Describe the bug**

Since `cvxpy 1.5.0` our of function (in [DIPY](https://github.com/dipy/dipy)) using `bmat` are failing.  Not sure what was the change and how to update the code.

all was working with 1.4.4.

**To Reproduce**

```python
import cvxpy as cp
import numpy.testing as npt
import numpy as np

def test_basic_test():
    V = np.arange(1, 7)[:, np.newaxis].astype(float)
    R = np.array(
        (
            [1, 4.24264069, 3.53553391],
            [4.24264069, 2, 2.82842712],
            [3.53553391, 2.82842712, 3],
        )
    )
    if V.shape[0] == 6:
        V = V.T

    f = 1 / np.sqrt(2)

    T = cp.bmat(
        [
            [V[0, 0], f * V[0, 5], f * V[0, 4]],
            [f * V[0, 5], V[0, 1], f * V[0, 3]],
            [f * V[0, 4], f * V[0, 3], V[0, 2]],
        ]
    )
    npt.assert_array_almost_equal(T.value, R)
```

**Expected behavior**

No errors like previous version of cvxpy.

**Output**

```python
        T = cp.bmat(
            [
                [V[0, 0], f * V[0, 5], f * V[0, 4]],
                [f * V[0, 5], V[0, 1], f * V[0, 3]],
                [f * V[0, 4], f * V[0, 3], V[0, 2]],
            ]
        )
>       npt.assert_array_almost_equal(T.value, R)

dipy/reconst/tests/test_qti.py:124: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/opt/homebrew/Caskroom/miniforge/base/envs/py310/lib/python3.10/site-packages/cvxpy/atoms/atom.py:357: in value
    return self._value_impl()
/opt/homebrew/Caskroom/miniforge/base/envs/py310/lib/python3.10/site-packages/cvxpy/atoms/atom.py:371: in _value_impl
    arg_val = arg._value_impl()
/opt/homebrew/Caskroom/miniforge/base/envs/py310/lib/python3.10/site-packages/cvxpy/atoms/atom.py:371: in _value_impl
    arg_val = arg._value_impl()
/opt/homebrew/Caskroom/miniforge/base/envs/py310/lib/python3.10/site-packages/cvxpy/atoms/atom.py:376: in _value_impl
    result = self.numeric(arg_values)
/opt/homebrew/Caskroom/miniforge/base/envs/py310/lib/python3.10/site-packages/cvxpy/atoms/atom.py:463: in new_numeric
    result = numeric_func(self, values)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = Expression(CONSTANT, NONNEGATIVE, (1,)), values = [array(1.0)]

    @AffAtom.numpy_numeric
    def numeric(self, values):
        """Reshape the value.
        """
>       return np.reshape(values[0], self.shape, self.order)
E       TypeError: reshape() takes from 1 to 2 positional arguments but 3 were given

/opt/homebrew/Caskroom/miniforge/base/envs/py310/lib/python3.10/site-packages/cvxpy/atoms/affine/reshape.py:93: TypeError
```

**Version**
 - OS: macOS
 - CVXPY Version: 1.5.3

**Additional context**

maybe related to #2534. all was working with cvxpy 1.4.4

**Comment by Transurgeon** [comment]
hi @skoudoro this is actually a NumPy issue that got propagated to cvxpy.
The fix should be done in #2542. Either you can downgrade NumPy to < 2.1 or try installing from cvxpy master. 
Let us know if this fixes the issue!

**Comment by skoudoro** [comment]
Thank you for your feedback. Closing,
